app/src/main/java/com/hedvig/app/authenticate/LoginDialog.kt

```diff
@@ -29,21 +29,23 @@ class LoginDialog : AuthenticateDialog() {
     }
   }
 
-  private fun bindNewStatus(state: AuthState): Any? = when (state) {
-    AuthState.INITIATED -> binding.authTitle.setText(R.string.BANK_ID_AUTH_TITLE_INITIATED)
-    AuthState.IN_PROGRESS -> binding.authTitle.setText(R.string.BANK_ID_LOG_IN_TITLE_IN_PROGRESS)
-    AuthState.UNKNOWN__,
-    AuthState.FAILED,
-    -> {
-      binding.authTitle.setText(R.string.BANK_ID_LOG_IN_TITLE_FAILED)
-      dialog?.setCanceledOnTouchOutside(true)
-    }
-    AuthState.SUCCESS -> {
-      binding.authTitle.setText(R.string.BANK_ID_LOG_IN_TITLE_SUCCESS)
-      dismissAllowingStateLoss()
-      startActivity(
-        LoggedInActivity.newInstance(requireContext(), withoutHistory = true),
-      )
+  private fun bindNewStatus(state: AuthState) {
+    when (state) {
+      AuthState.INITIATED -> binding.authTitle.setText(R.string.BANK_ID_AUTH_TITLE_INITIATED)
+      AuthState.IN_PROGRESS -> binding.authTitle.setText(R.string.BANK_ID_LOG_IN_TITLE_IN_PROGRESS)
+      is AuthState.UNKNOWN__,
+      AuthState.FAILED,
```

Comment:
Are we returning failed state anywhere now? Or will this status always be successful? Do we want some kind of time out in order to not get an infinite loading screen if we never get success?

Reply:
That's the issue though, we *are* getting a failed state but way too early as seen in the video in Jira. There's basically no case where we know that we got a Failed state and it's actually a failed state that we do want to show as an error from this subscription. I could not think of a better approach which doesn't also fail on the normal flow where the user simply might take too much time to fill in their data and backend randomly returns a failed state.

Reply:
The moment you see there that we get an error screen is when backend returned failed. But as seen in that video, the user didn't even do anything, just went inside the screen to fill in their data and then got kicked to an error screen.

Reply:
I see, so we are getting a failed state - but we cant trust it? Is this something that should be fixed on the backend instead?

Reply:
Could be yeah, I just couldn't find myself where this logic exists to try and see what may be going wrong. Let me try and think bout this a bit more actually.